Userland/Libraries/LibMarkdown/Document.cpp

```diff
@@ -15,20 +15,60 @@ namespace Markdown {
 String Document::render_to_html() const
 {
     StringBuilder builder;
+    builder.append(R"~~~(<!DOCTYPE html>
+<html>
+<head>
+    <style>
+        code { white-space: pre; }
+        .zoomable {
+            cursor: zoom-in;
+            max-width: 100%;
+        }
+        .zoomable.zoomed-in {
+            cursor: zoom-out;
+            max-width: none;
+        }
+    </style>
+    <script>
+        function imageClickEventListener(event) {
+            let image = event.target;
+            if (image.classList.contains("zoomable")) {
+                image.classList.toggle("zoomed-in");
+            }
+        }
+        function processImages() {
+            let images = document.querySelectorAll("img");
+            let windowWidth = window.innerWidth;
+            images.forEach((image) => {
+                if (image.naturalWidth > windowWidth) {
+                    image.classList.add("zoomable");
+                } else {
+                    image.classList.remove("zoomable");
+                    image.classList.remove("zoomed-in");
+                }
```

Comment:
Not a huge fan of outputting this into _any_ HTML generated from markdown... shrink-to-fit by default + having the user of the HTML insert a script into the webview that adds such customisation seems cleaner

Reply:
_Or_ a flag to optionally enable features such as zoomable images, code highlighting, etc which can be passed by Help, TextEditor, and whoever else is interested in a rich HTML document.
Ok, I think we have enough options 😄

Reply:
You're right, it's a bit gross given it's a user-side feature. Thank you for keeping my hack tendencies under control. :wink: